forc/src/ops/forc_fmt.rs

```diff
@@ -66,24 +66,23 @@ fn format_after_build(command: FormatCommand) -> Result<(), FormatError> {
                     }
                 }
             }
-
             // format manifest using taplo formatter
             if let Ok(file_content) = fs::read_to_string(&manifest_file) {
                 let taplo_alphabetize = taplo_fmt::Options {
                     reorder_keys: true,
                     ..Default::default()
                 };
                 let formatted_content = taplo_fmt::format(&file_content, taplo_alphabetize);
-                if command.check {
-                    if formatted_content != file_content {
-                        if !contains_edits {
-                            contains_edits = true;
-                        }
-                        eprintln!("\nManifest Forc.toml improperly formatted");
-                        display_file_diff(&file_content, &formatted_content)?;
-                    } else {
-                        format_file(&manifest_file, &formatted_content)?;
+                if !command.check {
+                    format_file(&manifest_file, &formatted_content)?;
+                } else if formatted_content != file_content {
+                    if !contains_edits {
+                        contains_edits = true;
                     }
+                    eprintln!("\nManifest Forc.toml improperly formatted");
+                    display_file_diff(&file_content, &formatted_content)?;
```

Comment:
If the command mode is `check`, we explicitly want to return a non-zero exit code back to the terminal. This will be handled automatically if we return an `Err` out of `main()`. Could you confirm that a non-zero exit code is being returned in the case where the toml is incorrectly formatted in `--check` mode?

Reply:
This is handled by the `contains_edit` boolean